registry/client.go

```diff
@@ -352,3 +352,118 @@ func (e Error) Error() string {
 
 	return "registry error: " + strconv.Itoa(e.StatusCode)
 }
+
+// Compatibility levels.
+const (
+	BackwardCL           string = "BACKWARD"
+	BackwardTransitiveCL string = "BACKWARD_TRANSITIVE"
+	ForwardCL            string = "FORWARD"
+	ForwardTransitiveCL  string = "FORWARD_TRANSITIVE"
+	FullCL               string = "FULL"
+	FullTransitiveCL     string = "FULL_TRANSITIVE"
+	NoneCL               string = "NONE"
+)
+
+/*
+ValidCompatibilityLevel returns whether or not a compatibility level
+is valid according to the ones described in:
+https://docs.confluent.io/platform/current/schema-registry/develop/api.html#compatibility
+.
+*/
+func ValidCompatibilityLevel(compatibilityLevel string) bool {
+	switch compatibilityLevel {
+	case BackwardCL:
+		return true
+	case BackwardTransitiveCL:
+		return true
+	case ForwardCL:
+		return true
+	case ForwardTransitiveCL:
+		return true
+	case FullCL:
+		return true
+	case FullTransitiveCL:
+		return true
+	case NoneCL:
+		return true
+	default:
+		return false
+	}
+}
+
+// compatibilityPayload is the response body obtained from the schema registry when updating a compatibility level.
+type compatibilityPayload struct {
+	Compatibility string `json:"compatibility"`
+}
+
+/*
+errByInvalidCL returns error for and invalid compatibility level, i.e. a compatibility
+level not present in:
+https://docs.confluent.io/platform/current/schema-registry/develop/api.html#compatibility .
+*/
+func errByInvalidCL(compatibilityLevel string) error {
+	return fmt.Errorf("invalid compatibility level %s", compatibilityLevel)
+}
```

Comment:
I find a function with a single line error very odd. I get the point of reuse, but this is odd none the less. It could perhaps be returned by the validation function instead.

Reply:
removed